### PR TITLE
Fix non-integer scale factor interpolation

### LINKED ISSUE #6836: Interp bilinear/bicubic use the wrong sampling scale for non-integer scale_factor (same as #3555, but for bilinear/bicubic)
Hit this converting a model with `F.interpolate(x, scale_factor=1.5, mode='bilinear')` — ncnn doesn't match PyTorch, but only for bilinear/bicubic. nearest is fine.

It's the same problem as #3555: when resizing by a scale factor (no explicit size), the coordinate scale should be `1/scale_factor`, not `w/outw` (input over rounded output). #3555 / 76e32e9 fixed that for **nearest**, but bilinear/bicubic still compute `w/outw` in `linear_coeffs`/`cubic_coeffs`. For integer scales the two are equal so it never shows up; for 1.5x on 7px, outw=10 and `7/10=0.7` vs `1/1.5=0.667` — every sample is offset.

Affects both front ends (PyTorch's default `recompute_scale_factor=False`, and ONNX Resize `half_pixel` — same as #3555).

Repro (pnnx, 3x7x7 fp32, vs PyTorch):

```
bilinear  scale=1.5  ac=False   max|diff| = 0.28
bicubic   scale=1.5  ac=False   max|diff| = 0.36
bilinear  scale=2.0  ac=False   0    (integer)
bilinear  scale=1.5  ac=True    0
nearest   scale=1.5             0
bilinear  size=(10,10)          0    (explicit size)
```

So it's bilinear/bicubic + non-integer scale + align_corners=False, no warning during conversion.

Fix is the same shape as 76e32e9 — use `1/width_scale` in the scale-factor path instead of `w/outw`. Explicit size stays `w/outw` (correct), integer scales stay bit-exact. I've got it working locally across the backends + a test; happy to send a PR.

### PR DESCRIPTION
**Pull request by xy200303**
## Summary
- Use the configured scale factor for bilinear and bicubic coordinate generation when output dimensions are inferred.
- Keep explicit output-size and size-expression behavior unchanged.
- Add regression coverage for 1.5x interpolation on a seven-pixel input.

## Verification
- Built and ran `test_interp` with the CPU-only CMake configuration.

## Notes
- Fixes #6836.

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=6844) <br/>All committers have signed the CLA.

File: src/layer/interp.cpp [hunk 1]
```diff
@@ -635,8 +635,8 @@ int Interp::forward(const std::vector<Mat>& bottom_blobs, std::vector<Mat>& top_
         float* alpha = (float*)(buf + outw + outh);           //new float[outw * 2];
         float* beta = (float*)(buf + outw + outh + outw * 2); //new float[outh * 2];
 
-        linear_coeffs(w, outw, xofs, alpha, align_corner);
-        linear_coeffs(h, outh, yofs, beta, align_corner);
+        linear_coeffs_impl(w, outw, xofs, alpha, align_corner, (output_width || !size_expr.empty()) ? w / (float)outw : 1.f / width_scale);
+        linear_coeffs_impl(h, outh, yofs, beta, align_corner, (output_height || !size_expr.empty()) ? h / (float)outh : 1.f / height_scale);
```
[the remark this answers is not in the document]
Comment (xy200303) [reply src/layer/interp.cpp]: Addressed in 433158f4. Dynamic target sizes now use the reference blob dimensions across the CPU paths, x86 BF16 paths, and Vulkan, including the 1D Vulkan path.

Added a deterministic 7x7-to-10x10 dynamic target size regression test for bilinear and bicubic interpolation. Built and ran `test_interp` in the default x86 Release configuration.

File: src/layer/interp.cpp [hunk 2]
```diff
@@ -535,7 +535,7 @@ int Interp::forward(const std::vector<Mat>& bottom_blobs, std::vector<Mat>& top_
             int* xofs = buf;
             float* alpha = (float*)(buf + outw);
 
-            linear_coeffs(w, outw, xofs, alpha, align_corner);
+            linear_coeffs_impl(w, outw, xofs, alpha, align_corner, (output_width || dynamic_target_size || !size_expr.empty()) ? w / (float)outw : 1.f / width_scale);
```
[the remark this answers is not in the document]
Comment (xy200303) [reply src/layer/interp.cpp]: Addressed in 32c71d8b. Inferred scale-factor resizes now bypass the identity fast path unless the corresponding scale factor is exactly 1, while explicit sizes, dynamic target sizes, and size expressions retain the fast path.

The scale and identity decisions are centralized in `Interp` helpers and used by the CPU and Vulkan backends. Added 1.1x 7-to-7 bilinear and bicubic regression coverage, including optimized-to-naive comparison. Built and ran `test_interp` in the default x86 Release configuration.

File: src/layer/interp.h [hunk 3]
```diff
@@ -22,6 +22,26 @@ class Interp : public Layer
...
+    }
+
+    float get_resize_scale_y(int h, int outh) const
+    {
+        return (output_height || dynamic_target_size || !size_expr.empty()) ? h / (float)outh : 1.f / height_scale;
+    }
+
+    bool is_identity_resize_x(int w, int outw) const
+    {
+        return outw == w && (output_width || dynamic_target_size || !size_expr.empty() || width_scale == 1.f);
```
[the remark this answers is not in the document]
Comment (xy200303) [reply src/layer/interp.h]: Addressed in 47a0f00f. Same-size small axes now remain on the identity path: length 1 for bilinear and lengths up to 3 for bicubic, matching the minimum source samples required by their coefficient paths.

The unified identity helper is used by every CPU and Vulkan backend. Added 1D and 3D small-axis regression coverage for inferred 1.1x scaling. Built and ran `test_interp` in the default x86 Release configuration.
